app/main.py

```diff
@@ -1,3 +1,57 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"{self.__class__.__name__}: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}(km={self.km})"
+
+    def __add__(self, other: int | float | Distance) -> Distance:
+        if isinstance(other, int | float):
+            return Distance(self.km + other)
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other: int | float | Distance) -> None:
+        if isinstance(other, int | float):
+            self.km += other
+            return self
+        self.km += other.km
+        return self
+
+    def __mul__(self, other: int | float) -> Distance:
+        return Distance(self.km * other)
```

Comment:
`other` can not be of type `Distance` in this method, because if you multiply km * km you get square km